src/Controllers/Table/FindReplaceController.php

```diff
@@ -99,16 +99,30 @@ public function __invoke(ServerRequest $request): void
             return;
         }
 
-        if (isset($_POST['find'])) {
-            $this->findAction();
+        $find = (string) $request->getParsedBodyParam('find');
+        $useRegex = (bool) $request->getParsedBodyParam('useRegex');
+        $replaceWith = (string) $request->getParsedBodyParam('replaceWith');
+        $columnIndex = (int) $request->getParsedBodyParam('columnIndex');
+
+        if (!empty($find)) {
+            $preview = $this->getReplacePreview($columnIndex, $find, $replaceWith, $useRegex, $this->connectionCharSet);
+            $this->response->addJSON('preview', $preview);
 
             return;
         }
 
         $this->addScriptFiles(['table/find_replace.js']);
 
-        if (isset($_POST['replace'])) {
-            $this->replaceAction();
+        if (!empty($request->getParsedBodyParam('replace'))) {
```

Comment:
Please don't use empty. We prefer stricter comparisons. In this case you can use `ServerRequest::hasBodyParam()`.

```suggestion
        if ($request->hasBodyParam('replace')) {
```

Reply:
Sure, makes sense, thanks!